src/lexer.rs

```diff
@@ -18,432 +20,614 @@ pub const fn is_alphabetic_extended(input: &char) -> bool {
   }
 }
 
-/// Lex an input string and returns [`Token`]s
-pub fn lex(input: &str, allow_trailing_operators: bool, default_degree: Unit) -> Result<Vec<Token>, String> {
-
-  let mut input = input.replace(",", ""); // ignore commas
+pub fn is_alphabetic_extended_str(input: &str) -> bool {
+  let x = match input {
+    value if value.chars().all(|c| ('a'..='z').contains(&c)) => true,
+    value if value.chars().all(|c| ('A'..='Z').contains(&c)) => true,
```

Comment:
Yeah that's a good call

Reply:
Actually, this breaks for cases like `2 watts plus 3 watts` because when we parse watt, we read the next word and check if it's `hour`. If it's not, we pass it to `parse_token()` to be re-used, which then would call `is_alphabetic_extended_str("plus")`.

I think we can get around it by calling `parse_word()` instead of `parse_token()` though

Reply:
Should be good now. Made it so when the `watt` match statement reads the next word, it calls `parse_word_if_non_empty()`, which means it's a no-op if the string is empty.